### ISSUE TITLE
[BUG]: Weird _data_type for artifact view

### ISSUE THREAD

**Issue by Finesim97** [zenml-io/zenml#166]
### What happened?

When trying to load my pandas DataFrame from an ArtifactReview, it tries to load: `env.lib.python3` and the `_data_type` attribute is set to `env.lib.python3.8.site-packages.pandas.core.frame.DataFrame`. I can't reproduce this behavior. 

While I try to recreate this in a new project, i am not able to see the same problem. 

```python3
pd.DataFrame
# pandas.core.frame.DataFrame
```

/env/lib/python3.8/site-packages is the site directory in my conda env, where the root directory seems to be my project directory and env my conda environment location directory.

This didn't occur in the 0.5.1 version.

### Reproduction steps

Not able to reproduce it yet

### ZenML Version

0.5.2

### Python Version

3.8

### OS Type

Linux

### Relevant log output

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_22166/2978891941.py in <module>
----> 1 x.read()

~/work/hda-solostar/env/lib/python3.8/site-packages/zenml/post_execution/artifact.py in read(self, output_data_type, materializer_class)
     90 
     91         if not output_data_type:
---> 92             output_data_type = source_utils.load_source_path_class(
     93                 self._data_type
     94             )

~/work/hda-solostar/env/lib/python3.8/site-packages/zenml/utils/source_utils.py in load_source_path_class(source)
    248         "load class from current repository state."
    249     )
--> 250     class_ = import_class_by_path(source)
    251     return class_
    252 

~/work/hda-solostar/env/lib/python3.8/site-packages/zenml/utils/source_utils.py in import_class_by_path(class_path)
    232     classname = class_path.split(".")[-1]
    233     modulename = ".".join(class_path.split(".")[0:-1])
--> 234     mod = importlib.import_module(modulename)
    235     return getattr(mod, classname)  # type: ignore[no-any-return]
    236 

~/work/hda-solostar/env/lib/python3.8/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

[...]

ModuleNotFoundError: No module named 'env.lib.python3'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by htahir1** [comment]
Thanks for the bug report. Are you currently experiencing this bug or were you able to solve it somehow?

**Comment by Finesim97** [comment]
Current problem, I am not aware of how ZenML detects paths to load, my conda env is inside my repo, but gitignored, may this be problem? I will try to use this setup for reproduction.

**Comment by schustmi** [comment]
Currently we distinguish between user modules and third-party packages based on the location of python files (if they are inside or outside the repo), so I believe your guess is right and the location of the conda env messes with our logic there.
Should be a relatively quick fix though, I'll look into it later.

https://github.com/zenml-io/zenml/blob/d742be60dbba5cf46ec8bc3e916a31c9ea3f1246/src/zenml/utils/source_utils.py#L214

**Comment by Finesim97** [comment]
I was able to reproduce the error, when the env is inside the repo. So you were right. Thank you so much for your time and this awesome project.

**Comment by htahir1** [comment]
Awesome @Finesim97 . @schustmi Please go ahead and close the issue if you dont have anything to add here

**Comment by schustmi** [comment]
@Finesim97 This issue should be fixed in #167. We'll probably release a new version friday next week which will include the fix
